### ISSUE TITLE
Incorrect autocorrection for `Style/SoleNestedConditional` cop

### ISSUE THREAD

**Issue by ymap** [rubocop/rubocop#12187]
The `Style/SoleNestedConditional` cop seems to unintentionally duplicate comments found at the end of a nested `if`, positioning them just before the outer `if`.

**Note:** It's worth noting that cases with solitary comments like `# noop` might emphasize the importance of this behavior anomaly.

Here's a demonstrative code snippet:

```ruby
if foo
  if bar
    # Comments.
  end
end
```

---
## Expected Behavior

The cop should maintain the integrity of the comments, resulting in:

```ruby
if foo && bar
  # Comments.
end
```

## Actual Behavior

```ruby
# Comments.
if foo && bar
  # Comments.
end
```

The comment appears to be duplicated.

## Steps to Reproduce the Problem

1. Save the provided code in a file named `example.rb`.

```ruby
if foo
  if bar
    # Comments.
  end
end
```

2. Execute `bundle exec rubocop -a example.rb`.

## RuboCop Version

```
$ bundle exec rubocop -V
1.56.2 (using Parser 3.2.2.3, rubocop-ast 1.29.0, running on ruby 3.0.2) [x86_64-linux]
  - rubocop-performance 1.19.0
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.23.2
```
